d3a_api_client/redis_client_base.py

```diff
@@ -66,15 +69,17 @@ def _subscribe_to_response_channels(self):
         channel_subs[f'{self._channel_prefix}/events/finish'] = self._on_finish
 
         self.pubsub.subscribe(**channel_subs)
-        self.pubsub.run_in_thread(daemon=True)
+        if pubsub is None:
+            self.pubsub.run_in_thread(daemon=True)
 
     def register(self, is_blocking=False):
         logging.info(f"Trying to register to {self.area_id} as client {self.client_id}")
         if self.is_active:
             raise RedisAPIException(f'API is already registered to the market.')
         data = {"name": self.client_id, "transaction_id": str(uuid.uuid4())}
+        with self.lock:
+            self._blocking_command_responses["register"] = data
```

Comment:
Why are you locking a `assigning to a dict` command? It is already blocking.

Reply:
As discussed offline, please remove the lock and change the order of assigning to _blocking_command_responses and publishing on other places too. Thanks